From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 16 Sep 2014 02:44:19 +0100
Subject: ip_ident: Avoid ABI change in 3.2.63

Commits 64b5c251d5b2 ('inetpeer: get rid of ip_id_count') and
04ca6973f7c1 ('ip: make IP identifiers less predictable'),
backported into 3.2.63, removed a field from struct inetpeer
and changed the parameters of __ip_select_ident().  Restore
them as unused padding.

--- a/include/net/inetpeer.h
+++ b/include/net/inetpeer.h
@@ -49,6 +49,7 @@ struct inet_peer {
 	union {
 		struct {
 			atomic_t			rid;		/* Frag reception counter */
+			atomic_t			ip_id_count;    /* now unused */
 			__u32				tcp_ts;
 			__u32				tcp_ts_stamp;
 		};
--- a/include/net/ip.h
+++ b/include/net/ip.h
@@ -265,7 +265,8 @@ int ip_dont_fragment(struct sock *sk, st
 }
 
 u32 ip_idents_reserve(u32 hash, int segs);
-void __ip_select_ident(struct iphdr *iph, int segs);
+void __ip_select_ident(struct iphdr *iph, struct dst_entry *dst /*unused*/,
+		       int segs);
 
 static inline void ip_select_ident_segs(struct sk_buff *skb, struct sock *sk, int segs)
 {
@@ -284,7 +285,7 @@ static inline void ip_select_ident_segs(
 			iph->id = 0;
 		}
 	} else {
-		__ip_select_ident(iph, segs);
+		__ip_select_ident(iph, NULL, segs);
 	}
 }
 
--- a/net/ipv4/route.c
+++ b/net/ipv4/route.c
@@ -1374,7 +1374,8 @@ u32 ip_idents_reserve(u32 hash, int segs
 }
 EXPORT_SYMBOL(ip_idents_reserve);
 
-void __ip_select_ident(struct iphdr *iph, int segs)
+void __ip_select_ident(struct iphdr *iph, struct dst_entry *dst /*unused*/,
+		       int segs)
 {
 	static u32 ip_idents_hashrnd __read_mostly;
 	static bool hashrnd_initialized = false;
